hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/operation/HederaCreateOperationExternalizer.java

```diff
@@ -92,6 +94,13 @@ public void externalize(MessageFrame frame, MessageFrame childFrame) {
             createdContractId = updater.idOfLastNewAddress();
         }
 
+        if (dynamicProperties.isContractsNoncesExternalizationEnabled()) {
+            final MutableAccount contractAccount = frame.getWorldUpdater()
+                    .getAccount(childFrame.getContractAddress())
+                    .getMutable();
+            updateContractNonces(createdContractId, contractAccount.getNonce());
```

Comment:
Probably we can double check if `childFrame.getContractAddress()` and `createdContractId` are the same?